internal/dataplane/kong_client_test.go

```diff
@@ -105,3 +118,179 @@ var (
 		Kind:    "Ingress",
 	}
 )
+
+// clientFactoryWithExpected implements ClientFactory interface and can be used
+// in tests to assert which clients have been created and signal failure if:
+// - client for an unexpected address gets created
+// - client which already got created was tried to be created second time.
+type clientFactoryWithExpected struct {
+	expected map[string]bool
+	t        *testing.T
+}
```

Comment:
How about adding a method that would assert that all expected calls were made, and calling it in every test? E.g. 
```go
func (c clientFactoryWithExpected) AssertExpectedCalls() {
  for addr, stillExpected := range c.expected {
    if stillExpected {
      c.t.Errorf("%s client expected to be called, but wasn't", addr)
    }
  }
}
```

Reply:
Nice idea, added.